javatests/arcs/core/crdt/CrdtSetTest.kt

```diff
@@ -144,6 +145,28 @@ class CrdtSetTest {
         ).isFalse()
     }
 
+    @Test
+    fun merge_reportsRemovalsFromMergedModel() {
+        val versionMap = VersionMap("alice" to 1)
+        val id = "added by alice removed by bob"
+
+        alice.applyOperation(Add("alice", versionMap, id))
+        bob.merge(alice.data)
+        assertThat(bob.data.values.keys).containsExactly(id)
+
+        val removalVersionMap = VersionMap("alice" to 1, "bob" to 0)
+        bob.applyOperation(Remove("bob", removalVersionMap, id))
+        val mergeChanges = alice.merge(bob.data)
+        assertThat(alice.data.values).isEmpty()
+        assertThat(mergeChanges.otherChange as CrdtChange.Operations).containsExactly(
```

Comment:
applying this op into bob won't make any difference right (the element has already been removed in bob model)? I think that's why we were not adding it to otherChange.
can you explain better what test is this particular change fixing and why?

Reply:
The operations that result from the merge are not for applying to the `bob` model, they are for announcing to the listeners of this store so that they can apply the removal change to their copies of the model.

At a higher level, this fixes a bug where if I have a handle for the same data in process A and process B, removing an item from the collection in process B did not result in any changes being reported in process A.

Without this, the tests:
`collection_removingFromA_isRemovedFromB` and `collection_removingFromA_isRemovedFromB` (in `HandleManagerTestBase.kt` will fail for the `DifferentHandleManagerDifferentStores` configuration.

@yuangu-google was also observing this issue in his on-device e2e tests.